### PR TITLE
Update docs for new version

### LINKED ISSUE #266: Temperature Anomalies - Apparent Temperature was 178,423°F
### Describe the bug

On multiple occasions recently, the apparent hi/lo temps have been inaccurate by a significant order of magnitude.

image

The issue first appeared on 2024-07-11 at 11:35:00 UTC.

### Expected behavior

I expect the weather forecast data to be within a reasonable range.

### Actual behavior

The weather forecast data said that my backyard was 10x hotter than the surface of the sun.

### API Endpoint

Production

### Location

Chicago, IL, US

### Other details

_No response_

### Troubleshooting steps

- [X] I have searched this repository and [Home Assistant Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the issue has already been reported.
- [X] I have read through the [API documentation](https://pirateweather.net/en/latest/API/) before opening this issue.
- [X] I have written an informative title.

### LINKED ISSUE #283: Pressure has extreme values (like -308024647.68 or 151026579537.92) at some point of the forecast
### Describe the bug

A pressure runs into extreme positive and negative values (so large that my uint32 encoding fails).
I can see it happening right now (timestamp 1723022370 or `2024-08-07T09:19:30+00:00`)

Request: https://api.pirateweather.net/forecast/<api token>/38.631,-105.071?exclude=currently,minutely,daily,alerts,flags&extend=hourly&units=si&version=2

Also visible on Merry Sky: https://merrysky.net/forecast/38.631,-105.071/ca

image

### Expected behavior

The pressure should have expected values

### Actual behavior

The pressure have values exceeding uint32 in both positive and negative values at some point in the forecast

### API Endpoint

Production

### Location

38.631,-105.071

### Other details

_No response_

### Troubleshooting steps

- [X] I have searched this repository and [Home Assistant Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the issue has already been reported.
- [X] I have read through the [API documentation](https://pirateweather.net/en/latest/API/) before opening this issue.
- [X] I have written an informative title.

### LINKED ISSUE #284: Pressure reading jumps to 3000 mbar at midnight
### Describe the bug

I'm using Home Assistant but there have been no changes that I know of in that area. Since a week or so the pressure value jumps exactly at midnight to 3000 and then it recovers.

image

### Expected behavior

Pressure should have a reasonable value.

### Actual behavior

Pressure jumps to 3000 mbar

### API Endpoint

Production

### Location

Timisoara, Romania

### Other details

_No response_

### Troubleshooting steps

- [X] I have searched this repository and [Home Assistant Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the issue has already been reported.
- [X] I have read through the [API documentation](https://pirateweather.net/en/latest/API/) before opening this issue.
- [X] I have written an informative title.

### PR DESCRIPTION
**Pull request by cloneofghosts**
@alexander0042 Just getting ahead of the curve and create a branch where you can document the changes coming in the next release. The only thing changed so far is the version number in the OpenAPI spec.

This release should

* Fix #266 .
* Fix #283 
* Fix #284 
* It also should address the issues seen in #229 

Things to update before this can be merged in:

- [x] Changelog
- [x] API documentation (if needed)
- [x] OpenAPI spec (if needed)
- [x] Recent updates section on the homepage
- [x] Roadmap (if needed)

### CODE REVIEW

**Comment by alexander0042** [comment]
Great idea to start pulling this together! Just released v2.1 live on the dev endpoint, and I want to see how it does for a few days, but so long as it's stable it should be ready for prime time

**Comment by cloneofghosts** [comment]
Feel free to update the docs on that branch for the new update.

I just did a quick test on the dev endpoint and I noticed that `pressure` and apparent temperature are the same as the production endpoint. Are you still working on those fixes or they just get missed?

**Comment by cloneofghosts** [comment]
@alexander0042 Feel free to update the docs for anything related to the update. I also added a v2.1 section to the changelog so you can start adding the changes there.

When I was testing the development server there were some issues which I thought were going to be fixed in this version but don't seem to be fixed. Namely these issues:

* #269 
* #268 
* #265 
* #246 

Is it just the case of the version on dev currently doesn't have these fixes or are they not going to be included in v2.1 and will come afterwards?

**Comment by cloneofghosts** [comment]
I noticed there were a few things in the repo which weren't in the roadmap so I went and added them. You can always remove if you don't want them there are re-order things if needed.

**Comment by cloneofghosts** [comment]
@alexander0042 This PR should be good to review now as everything has been updated. In the changelog I had initially marked #269, #268, #265 and #246 as being fixed but considering the issues that have popped up today I think 2.1 can be launched and the other issues be fixed in a 2.1.1 version.

The release date of 2.1 still needs to be added.

**Comment by alexander0042** [comment]
Thanks for the ping! I'm always tempted to try to get everything in one big release, but you make a great point and it's better just to push this out to fix the data stability issues and then go from there. Data ingest has been a whole can of worms, but I really think this approach will scale for the future, and will also make it much easier for people to eventually run it on their own!

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ To sign up, a free API key can be requested at [https://pirateweather.net/](http
 ## Why
 
 1. Weather forecasts are produced by government agencies using computational models; however, the data is hard to work with. Instead, there are a bunch of companies that take this data, process it, and provide easier to work with APIs. However, the steps that the data goes through aren't public or documented (for example, what do they mean by "percent of precipitation"?!?). I put this service together in order to provide a better way for people to know where their weather data is coming from and how it is being processed! 
-2. The Dark Sky API is great, and widely used. Unfortunately, it will shut down at the end of 2022, so in order to keep legacy services operating, I wanted to put a service together that would be drop-in compatible
+2. The Dark Sky API is great, and widely used. Unfortunately, it shut down at the end of 2022, so in order to keep legacy services operating, I wanted to put a service together that would be drop-in compatible
```
Comment (alexander0042) [inline README.md:9]: RIP Dark Sky

**Review by alexander0042 — APPROVED** [review]

**Review by alexander0042 — APPROVED** [review]

**Comment by cloneofghosts** [comment]
> Thanks for the ping! I'm always tempted to try to get everything in one big release, but you make a great point and it's better just to push this out to fix the data stability issues and then go from there. Data ingest has been a whole can of worms, but I really think this approach will scale for the future, and will also make it much easier for people to eventually run it on their own!

As much as a big release would be nice imo any stability issues should get pushed out as soon as their ready. For those other issues it would be nice if they were fixed I'd rather the stability issues be fixed first and have those other issues be fixed afterwards.

**Pull request merged**